src/main/java/com/google/api/codegen/discovery/config/ApiaryConfigToSampleConfigConverter.java

```diff
@@ -141,18 +143,23 @@ private MethodInfo createMethod(Method method) {
    */
   private FieldInfo createFieldInfo(Field field, Type containerType, Method method) {
     String placeholder = "";
+    boolean isPlaceholderSingular = true;
     TypeInfo typeInfo = createTypeInfo(field, method);
     if (typeInfo.kind() == Field.Kind.TYPE_STRING && !typeInfo.isArray() && !typeInfo.isMap()) {
       String pattern = apiaryConfig.getFieldPattern().get(containerType.getName(), field.getName());
       if (!Strings.isNullOrEmpty(pattern)) {
-        placeholder = DefaultString.getPlaceholder(field.getName(), pattern);
+        placeholder = DefaultString.getNonTrivialPlaceholder(pattern);
+        // String placeholders are always contained within braces, so counting
+        // the number of open braces is an easy test for singularity.
+        isPlaceholderSingular = StringUtils.countMatches(placeholder, '{') < 2;
```

Comment:
Can't happen as far as I know... there shouldn't be a pattern where there's nothing to replace, at least I haven't seen such a field.

Also, the DefaultString algorithm only operates on a specific pattern type, so if there's nothing to replace in the pattern it does no work

So the scenario you describe should never happen